fitnesse/resources/bootstrap-plus/js/bootstrap-plus.js

```diff
@@ -216,5 +222,109 @@ $( document ).ready(function() {
+        }
+    );
+    // Add hover function to test page to show add tag button
+    $('.test').parent().hover(
+        function() {
+            $(this).find('.addTag:first').css("visibility", "visible");
+        }, function() {
+            $(this).find('.addTag:first').css("visibility", "hidden");
+        }
+    );
+
+    // Click add tag function
+    $('.addTag').click(function() {
+        addTagInput($(this));
+    });
 });
 
+function addTagInput(currentAddTagButton){
+    //Remove all existing tag input fields
+    $('.tagInputOverview').remove();
+
+    //Add input field
+    $(currentAddTagButton).after('<input type="text" class="tagInputOverview">');
+
+    //Add focus after clicking button
+    $('.tagInputOverview').focus();
+
+    $('.tagInputOverview').keyup(function(event) {
+        //If "Enter" button is pressed
+        if (event.keyCode == 13) {
+            //Get current input value & replace empty spaces at the end of input
+            const inputValue = $('.tagInputOverview').val().replace(/\s+\S*$/, "");
+            //Get href value of the a tag
+            const currentURL = $(currentAddTagButton).siblings('a').attr('href');
+            //Call get current tag list function
+            GetCurrentTagList(currentURL, inputValue);
+        }
+    });
+}
+
+//Get current tag list function
+function GetCurrentTagList(currentURL, newTags){
+    //Get current tag list
+    $.ajax({
+        type: 'POST',
+        url: "http://localhost:9090/" + currentURL,
+        contentType: 'application/json; charset=utf-8',
+        data : 'responder=tableOfContents',
+        dataType: 'json',
+        success: function(data){
+            //Convert data object to string
+            const currentTagList = data[0].tags.toString();
+            //Convert input tags to lowercase
+            const lowerCaseTags = newTags.toLowerCase();
+            //Check if there are any tags currently present
+            if (currentTagList.length > 0){
+                //Check if input tag exists in current tag list
+                const checkIfExists = currentTagList.includes(lowerCaseTags);
+                //If tag doesn't exist yet, post it
+                if (checkIfExists === false){
+                    //Combine the current tag list and the input tag(s) in 1 variable
+                    const newTagList = currentTagList + ", " + lowerCaseTags;
+                    //Send current href value, new tag list and input tag(s) to post tag function
+                    postTag(currentURL, newTagList, newTags);
+                }
+            }
+            //If there are no tags present only post the input tags
+            else {
+                postTag(currentURL, lowerCaseTags, newTags);
+            }
+        }
+    });
+}
+
+//Post new tag list function
+function postTag(currentURL, tagList, inputTag) {
+    $.ajax({
+        type: 'POST',
+        url: "http://localhost:9090/" + currentURL,
```

Comment:
Het url moet runtime worden bepaald. de poort (9090) is door de gebruiker te configureren. Er zijn ook setups waar fitnesse draait op een andere server dan lokaal. je kunt in vanilla js volgens mij gewoon `location.host` gebruiken om hostnaam:port te krijgen.

Reply:
Hostnaam en port variabel gemaakt m.b.v. `location.host`.